common/tier4_automatic_goal_rviz_plugin/src/automatic_goal_sender.hpp

```diff
@@ -0,0 +1,174 @@
+  {
+    if (!client->service_is_ready()) {
+      RCLCPP_WARN(get_logger(), "SetRoutePoints client is unavailable");
+      return false;
+    }
+
+    auto req = std::make_shared<SetRoutePoints::Request>();
+    req->header = goals_list_.at(goal_index)->header;
+    req->goal = goals_list_.at(goal_index)->pose;
+    client->async_send_request(
+      req, [this](typename rclcpp::Client<SetRoutePoints>::SharedFuture result) {
+        if (result.get()->status.code != 0) state = State::ERROR;
+        printCallResult<SetRoutePoints>(result);
+        onCallResult();
+      });
+    return true;
+  }
+  template <typename T>
+  bool callService(const typename rclcpp::Client<T>::SharedPtr client)
+  {
+    if (!client->service_is_ready()) {
+      RCLCPP_WARN(get_logger(), "Client is unavailable");
+      return false;
+    }
+
+    auto req = std::make_shared<typename T::Request>();
+    client->async_send_request(req, [this](typename rclcpp::Client<T>::SharedFuture result) {
+      if (result.get()->status.code != 0) state = State::ERROR;
+      printCallResult<T>(result);
+      onCallResult();
+    });
+    return true;
+  }
+  template <typename T>
+  void printCallResult(typename rclcpp::Client<T>::SharedFuture result)
+  {
+    if (result.get()->status.code != 0) {
+      RCLCPP_ERROR(
+        get_logger(), "Service type \"%s\" status: %d, %s", typeid(T).name(),
+        result.get()->status.code, result.get()->status.message.c_str());
+    } else {
+      RCLCPP_DEBUG(
+        get_logger(), "Service type \"%s\" status: %d, %s", typeid(T).name(),
+        result.get()->status.code, result.get()->status.message.c_str());
+    }
+  }
+
+  // Update
+  void updateGoalsList();
+  virtual void updateAutoExecutionTimerTick();
+
+  // File
+  void loadGoalsList(const std::string & file_path);
+  void updateAchievedGoalsFile(const unsigned goal_index);
+  void resetAchievedGoals();
+  std::string getTimestamp() const
+  {
+    char buffer[128];
+    std::time_t now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    std::strftime(&buffer[0], sizeof(buffer), "%Y-%m-%d %H:%M:%S", std::localtime(&now));
+    return std::string(buffer);
+  }
+
+  // Sub
+  void onRoute(const RouteState::ConstSharedPtr msg);
+  void onOperationMode(const OperationModeState::ConstSharedPtr msg);
+
+  // Interface
+  virtual void onRouteUpdated(const RouteState::ConstSharedPtr) {}
+  virtual void onOperationModeUpdated(const OperationModeState::ConstSharedPtr) {}
+  virtual void onCallResult() {}
+  virtual void onGoalListUpdated() {}
+
+  // Cli
+  rclcpp::Client<ChangeOperationMode>::SharedPtr cli_change_to_autonomous_;
+  rclcpp::Client<ChangeOperationMode>::SharedPtr cli_change_to_stop_;
+  rclcpp::Client<ClearRoute>::SharedPtr cli_clear_route_;
+  rclcpp::Client<SetRoutePoints>::SharedPtr cli_set_route_;
+
+  // Containers
+  unsigned current_goal{0};
+  State state{State::INITIALIZING};
```

Comment:
this current_goal and state should  be current_goal_ and state_  and please fix other part as well.